File: Grind169/Week1/121-Best-Time-to-Buy-and-Sell-Stock/solution.cpp
#include <vector>
using namespace std;

class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_price = 0;

        int left=0;
        int right=1;

        while (right<prices.size()){
            int curr_profit=prices[right]-prices[left];

            if (prices[left]<prices[right]){
                max_price=max(max_price,curr_profit);
            } else {
                left=right;
            }

            right++;
        }

        return max_price;

    }
};

File: Grind169/Week1/121-Best-Time-to-Buy-and-Sell-Stock/solution.py
from typing import List


class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # use two pointer
        max_price = 0

        left = 0
        right = 1

        while right < len(prices):
            curr_profit = prices[right] - prices[left]

            if prices[right] > prices[left]:
                max_price = max(max_price, curr_profit)
            else:
                left = right

            right += 1

        return max_price
